turbopack/crates/turbopack-ecmascript/tests/tree-shaker/analyzer/effects-1/output.md

```diff
@@ -0,0 +1,763 @@
+    Item6 --> Item1;
+    Item7 --> Item5;
+    Item7 --> Item1;
+    Item8 --> Item6;
+    Item8 --> Item7;
+    Item8 --> Item1;
+    Item9 --> Item8;
+    Item9 --> Item1;
+    Item9 -.-> Item6;
+    Item9 -.-> Item7;
+    Item10 --> Item9;
+    Item10 --> Item1;
+    Item11 --> Item10;
+    Item11 --> Item1;
+    Item11 --> Item3;
+    Item12 --> Item10;
+    Item12 --> Item1;
+    Item13 --> Item10;
+    Item13 --> Item1;
+    Item13 --> Item12;
+    Item13 -.-> Item11;
+    Item14 -.-> Item13;
+    Item14 --> Item1;
+    Item15 --> Item11;
+```
+# Final
+```mermaid
+graph TD
+    N0["Items: [ItemId(ModuleEvaluation)]"];
+    N1["Items: [ItemId(0, VarDeclarator(0))]"];
+    N2["Items: [ItemId(1, Normal)]"];
+    N3["Items: [ItemId(2, Normal)]"];
+    N4["Items: [ItemId(3, Normal)]"];
+    N5["Items: [ItemId(4, Normal)]"];
+    N6["Items: [ItemId(5, VarDeclarator(0))]"];
+    N7["Items: [ItemId(6, VarDeclarator(0))]"];
+    N8["Items: [ItemId(7, Normal)]"];
+    N9["Items: [ItemId(8, Normal)]"];
+    N10["Items: [ItemId(9, Normal)]"];
+    N11["Items: [ItemId(10, Normal)]"];
+    N12["Items: [ItemId(11, VarDeclarator(0))]"];
+    N13["Items: [ItemId(12, Normal)]"];
+    N14["Items: [ItemId(13, Normal)]"];
+    N2 --> N1;
+    N3 --> N1;
+    N3 --> N2;
+    N4 -.-> N3;
+    N4 --> N1;
+    N5 -.-> N3;
+    N5 --> N1;
+    N6 --> N5;
+    N6 --> N1;
+    N7 --> N5;
+    N7 --> N1;
+    N8 --> N6;
+    N8 --> N7;
+    N8 --> N1;
+    N9 --> N8;
+    N9 --> N1;
+    N9 -.-> N6;
+    N9 -.-> N7;
+    N10 --> N9;
+    N10 --> N1;
+    N11 --> N10;
+    N11 --> N1;
+    N11 --> N3;
+    N12 --> N10;
+    N12 --> N1;
+    N13 --> N10;
+    N13 --> N1;
+    N13 --> N12;
+    N13 -.-> N11;
+    N14 -.-> N13;
+    N14 --> N1;
+    N0 --> N11;
+```
+# Entrypoints
+
+```
+{
+    ModuleEvaluation: 0,
+    Exports: 15,
+}
+```
+
+
+# Modules (dev)
+## Part 0
+```js
+import "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 11
+};
+"module evaluation";
+
+```
+## Part 1
+```js
+let x = 0;
+export { x as a } from "__TURBOPACK_VAR__" assert {
+    __turbopack_var__: true
+};
+
+```
+## Part 2
+```js
+import { a as x } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: -1
+};
+console.log(x);
+
+```
+## Part 3
+```js
+import { a as x } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: -1
+};
+import "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 2
+};
+console.log(x);
+
+```
+## Part 4
+```js
+import { a as x } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: -1
+};
+import "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 3
+};
+x = 1;
+
+```
+## Part 5
+```js
+import { a as x } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: -1
+};
+import "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 3
+};
+x = 2;
+
+```
+## Part 6
+```js
+import { a as x } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: -1
+};
+import "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 5
```

Comment:
A read should depend on `last_reads.iter()`, but 4 is removed from the `last_reads` whille processing 5